### DISCUSSION TITLE
Target: serverless still required react, react-dom to be installed

### DISCUSSION THREAD

**Discussion by grepug** [vercel/next.js#11459] [Help]
# Bug report

## Describe the bug

`next start` a serverless build, still prompt me the error that require 'react', 'react-dom'.

I thought the React is inlined with the output, isn't it?

**Comment by timneutkens** [comment]
It's inlined in the bundle indeed.

You shouldn't use `next start` to production host the `serverless` target, it's not optimized to run everything as a single "server" instead the bundles should be wrapped and served separately

**Comment by grepug** [comment]
I was trying to deploy the serverless build to AliCloud which has its own implementation for Serverless where `res` and `req` objects are slightly different with `next`'s output.

Am I supposed to write a custom server to hook them up? or any good idea?
